huaweicloud/services/vod/resource_huaweicloud_vod_transcoding_template_group.go

```diff
@@ -0,0 +1,576 @@
+		d.Set("common", flattenCommon(templateGroup.Common)),
+		d.Set("quality_info", flattenQualityInfoList(templateGroup.QualityInfoList)),
+	)
+
+	if err = mErr.ErrorOrNil(); err != nil {
+		return diag.Errorf("error setting VOD transcoding template group fields: %s", err)
+	}
+
+	return nil
+}
+
+func resourceTranscodingTemplateGroupUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	config := meta.(*config.Config)
+	client, err := config.HcVodV1Client(config.GetRegion(d))
+	if err != nil {
+		return diag.Errorf("error creating VOD client : %s", err)
+	}
+
+	updateOpts := vod.ModifyTransTemplateGroup{
+		GroupId:              d.Id(),
+		Name:                 d.Get("name").(string),
+		Description:          utils.String(d.Get("description").(string)),
+		WatermarkTemplateIds: utils.ExpandToStringListPointer(d.Get("watermark_template_ids").([]interface{})),
+		Common:               buildCommonOpts(d.Get("common").(([]interface{}))),
+		QualityInfoList:      buildQualityInfoListOpts(d.Get("quality_info").(([]interface{}))),
+	}
+
+	if d.Get("auto_encrypt").(bool) {
+		updateOpts.AutoEncrypt = utils.Int32(int32(1))
+	}
+
+	var status vod.ModifyTransTemplateGroupStatus
+	if d.Get("is_default").(bool) {
+		status = vod.GetModifyTransTemplateGroupStatusEnum().E_1
+	} else {
+		status = vod.GetModifyTransTemplateGroupStatusEnum().E_0
+	}
+	updateOpts.Status = &status
+	log.Printf("[DEBUG] Update VOD transcoding template group Options: %#v", updateOpts)
+
+	updateReq := vod.UpdateTemplateGroupRequest{
+		Body: &updateOpts,
+	}
+
+	_, err = client.UpdateTemplateGroup(&updateReq)
+	if err != nil {
+		return diag.Errorf("error updating VOD transcoding template group: %s", err)
+	}
+
+	return resourceTranscodingTemplateGroupRead(ctx, d, meta)
+}
+
+func resourceTranscodingTemplateGroupDelete(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	config := meta.(*config.Config)
+	client, err := config.HcVodV1Client(config.GetRegion(d))
+	if err != nil {
+		return diag.Errorf("error creating VOD client : %s", err)
+	}
+
+	_, err = client.DeleteTemplateGroup(&vod.DeleteTemplateGroupRequest{GroupId: d.Id()})
+	if err != nil {
+		return diag.Errorf("error deleting VOD transcoding template group: %s", err)
+	}
+
+	return nil
+}
+
+func flattenCommon(common *vod.Common) []map[string]interface{} {
+	if common == nil {
+		return nil
+	}
+
+	var lowBitrateHd bool
+	if common.Pvc == vod.GetCommonPvcEnum().E_1 {
+		lowBitrateHd = true
+	}
+
+	var videoCodec string
+	if *common.VideoCodec == vod.GetCommonVideoCodecEnum().H265 {
+		videoCodec = "H265"
+	} else {
+		videoCodec = "H264"
+	}
+
+	var audioCodec string
+	switch *common.AudioCodec {
+	case vod.GetCommonAudioCodecEnum().HEAAC1:
+		audioCodec = "HEAAC1"
+	case vod.GetCommonAudioCodecEnum().HEAAC2:
+		audioCodec = "HEAAC2"
+	case vod.GetCommonAudioCodecEnum().MP3:
+		audioCodec = "MP3"
+	default:
+		audioCodec = "AAC"
```

Comment:
maybe defaults to empty is better

Reply:
just ok, think about the import operation.

Reply:
fixed